test/data_processing/test_nodes.py

```diff
@@ -301,3 +302,29 @@ def test_probability_balanced(self):
         data = {"1": 512, "0": 512}
         processed_data = node(data=np.asarray([data]))
         self.assertAlmostEqual(unp.nominal_values(processed_data), 0.5)
+
+
+class TestRestless(QiskitExperimentsTestCase):
+    """Test the restless measurements node."""
+
+    def test_restless_classify(self):
+        """Test the classification of restless shots."""
+        previous_shot = "11000110"
+        shot = "11100010"
+
+        restless_classified_shot = RestlessToCounts._restless_classify(shot, previous_shot)
+        self.assertAlmostEqual(restless_classified_shot, "00100100")
+
+    def test_restless_process(self):
+        """Test if a restless memory is correctly post-processed."""
+        node = RestlessToCounts(header={"memory_slots": 1})
+
+        data = [["0x1", "0x0"], ["0x1", "0x1"]]
```

Comment:
Can we extend this to the case of 2 gates and 4 shots? E.g. X and SX?

Reply:
See 546412a.